src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -47,6 +47,12 @@ internal static void CopyProperties(this PreferenceSettings source, PreferenceSe
     /// </summary>
     public class PreferenceSettings : NotificationObject, IPreferences, IRenderPrecisionPreference, IDisablePackageLoadingPreferences, ILogSource, IHideAutocompleteMethodOptions
     {
+        internal readonly static Lazy<PreferenceSettings>
+            lazy = new Lazy<PreferenceSettings>
+            (() => PreferenceSettings.Load(PathManager.Instance.PreferenceFilePath));
+         
+        public static PreferenceSettings Instance { get { return lazy.Value; } }
```

Comment:
I feel like this is going to do weird things to the tests.

Reply:
I would say this change in behavior of a public class, potentially the same for the PathManager is the bigger API break in this PR...

Reply:
Is it possible to still create more than one instance of these class, if you don't access the Instance property?

Reply:
> Is it possible to still create more than one instance of these classes, if you don't access the Instance property?

yes, that's possible